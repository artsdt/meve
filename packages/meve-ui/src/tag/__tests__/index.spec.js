import Tag from '..'
import { mount } from '@vue/test-utils'

test('test tag type', () => {
  const wrapper = mount({
    components: {
      [Tag.name]: Tag,
    },
    template: `
      <div>
        <m-tag>hello</m-tag>
        <m-tag type="primary">hello</m-tag>
        <m-tag type="warning">hello</m-tag>
        <m-tag type="info">hello</m-tag>
        <m-tag type="success">hello</m-tag>
        <m-tag type="error">hello</m-tag>
      </div>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test tag size', () => {
  const wrapper = mount({
    components: {
      [Tag.name]: Tag,
    },
    template: `
      <div>
        <m-tag size="mini">hello</m-tag>
        <m-tag size="small">hello</m-tag>
        <m-tag>hello</m-tag>
        <m-tag size="large">hello</m-tag>
      </div>
    `,
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test tag trigger click and touchstart', async () => {
  const wrapper = mount(Tag)
  await wrapper.trigger('click')
  await wrapper.trigger('touchstart')
  expect(wrapper.emitted().click).toBeTruthy()
  expect(wrapper.emitted().touchstart).toBeTruthy()
})

test('test tag disabled', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      disabled: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })
  await wrapper.trigger('click')
  await wrapper.trigger('touchstart')
  expect(wrapper.emitted().click).toBeFalsy()
  expect(wrapper.emitted().touchstart).toBeFalsy()
  expect(wrapper.html()).toMatchSnapshot()
})

test('test tag block', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      block: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test tag text and outline', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      text: true,
      outline: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()
})

test('test tag closeable', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      closeable: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.m-button').trigger('click')
  expect(wrapper.emitted().close).toBeTruthy()
})

test('test tag closeable with disabled', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      disabled: true,
      closeable: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.m-button').trigger('click')
  expect(wrapper.emitted().click).toBeFalsy()
  expect(wrapper.emitted().close).toBeFalsy()
})

test('test tag closeable with readonly', async () => {
  const wrapper = mount(Tag, {
    propsData: {
      readonly: true,
      closeable: true,
    },
    scopeSlots: {
      default: () => 'hello',
    },
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.m-button').trigger('click')
  expect(wrapper.emitted().click).toBeTruthy()
  expect(wrapper.emitted().close).toBeFalsy()
})
