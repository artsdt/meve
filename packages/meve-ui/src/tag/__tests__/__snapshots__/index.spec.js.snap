// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test tag block 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--flex m-tag--block m-tag--default\\">
  <div class=\\"m-tag__content\\"></div>
</div>"
`;

exports[`test tag closeable 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default\\">
  <div class=\\"m-tag__content\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--text-default m-button--text m-button--round m-tag__close-button\\" tag-cover=\\"true\\">
      <div class=\\"m-button__content\\"><i tag-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-close m-tag__close-icon\\"></i></div>
    </button></div>
</div>"
`;

exports[`test tag closeable with disabled 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default m-tag--disabled\\">
  <div class=\\"m-tag__content\\"><button tabindex=\\"-1\\" disabled=\\"disabled\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--text-default m-button--text m-button--round m-button--text-disabled m-tag__close-button\\" tag-cover=\\"true\\">
      <div class=\\"m-button__content\\"><i tag-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-close m-tag__close-icon\\"></i></div>
    </button></div>
</div>"
`;

exports[`test tag closeable with readonly 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default\\">
  <div class=\\"m-tag__content\\"><button tabindex=\\"-1\\" class=\\"m-button m--box m-button--normal m--inline-flex m-button--text-default m-button--text m-button--round m-tag__close-button\\" tag-cover=\\"true\\">
      <div class=\\"m-button__content\\"><i tag-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-close m-tag__close-icon\\"></i></div>
    </button></div>
</div>"
`;

exports[`test tag disabled 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default m-tag--disabled\\">
  <div class=\\"m-tag__content\\"></div>
</div>"
`;

exports[`test tag size 1`] = `
"<div>
  <div class=\\"m-tag m--box m-tag--mini m--inline-flex m-tag--default\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--small m--inline-flex m-tag--default\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--large m--inline-flex m-tag--default\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
</div>"
`;

exports[`test tag text and outline 1`] = `
"<div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--text-default m-tag--text m-tag--outline\\">
  <div class=\\"m-tag__content\\"></div>
</div>"
`;

exports[`test tag type 1`] = `
"<div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--default\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--primary\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--warning\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--info\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--success\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
  <div class=\\"m-tag m--box m-tag--normal m--inline-flex m-tag--error\\">
    <div class=\\"m-tag__content\\">hello</div>
  </div>
</div>"
`;
